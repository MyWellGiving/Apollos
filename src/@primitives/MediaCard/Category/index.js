import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';

import withTheme from '@primitives/withTheme';
import styled from '@primitives/styled';
import { H7 } from '@primitives/typography';
import Icon from '@primitives/Icon';
import rem from '@utils/remUnit';

const enhance = compose(
  pure,
  withTheme(),
  setPropTypes({
    type: PropTypes.string,
    color: PropTypes.string,
  }),
);

const Wrapper = styled(({ theme }) => ({
  flexDirection: 'row',
  alignItems: 'center',
  flex: 1,
  paddingTop: theme.baseUnit / 2,
  paddingHorizontal: theme.baseUnit,
  paddingBottom: theme.baseUnit,
}))(View);

const StyledH7 = styled(({ theme, fontColor }) => ({
  paddingHorizontal: theme.baseUnit / 2,
  color: fontColor,
}))(H7);

const Category = enhance(({
  type,
  color,
  theme,
}) => (
  <Wrapper>
    <Icon
      name={'video'}
      size={rem(1.2, theme)}
      fill={color}
    />
    <StyledH7 fontColor={color}>{type}</StyledH7>
  </Wrapper>
));

export default Category;
