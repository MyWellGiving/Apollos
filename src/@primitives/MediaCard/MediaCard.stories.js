import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import MediaCard from './';

storiesOf('@primitives/MediaCard', module)
  .add('Default', () => {
    const centered = {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
      backgroundColor: '#f7f7f7',
    };

    const cardDimensions = {
      height: 400,
      width: '92%',
    };

    return (
      <View style={centered}>
        <MediaCard
          style={cardDimensions}
          image={'https://picsum.photos/600/400/?random'}
          title={'Boom'}
          mediaType={'What'}
        />
      </View>
    );
  });
