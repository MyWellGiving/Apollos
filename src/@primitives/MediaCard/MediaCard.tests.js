import React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '@primitives/ThemeProvider';
import MediaCard from './';

describe('the Card component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <MediaCard
          title={'Boom'}
          image={'https://picsum.photos/600/400/?random'}
          category={'What'}
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const cardDimensions = {
      height: 400,
      width: '92%',
    };

    const tree = renderer.create(
      <ThemeProvider>
        <MediaCard style={cardDimensions} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
