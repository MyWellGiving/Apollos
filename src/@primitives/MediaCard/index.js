import React from 'react';
import {
  Platform,
  View,
  Image,
  TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import {
  compose,
  pure,
  setPropTypes,
} from 'recompose';
import { startCase, toLower } from 'lodash';

import withTheme from '@primitives/withTheme';
import styled from '@primitives/styled';
import { H4 } from '@primitives/typography';
import Icon from '@primitives/Icon';
import rem from '@utils/remUnit';

import Category from './Category';

const enhance = compose(
  pure,
  withTheme(),
  setPropTypes({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    cardColor: PropTypes.string,
    style: View.propTypes.style,
  }),
);

const CardWrapper = styled(({ theme }) => ({
  marginHorizontal: theme.baseUnit / 2,
  marginVertical: theme.baseUnit / 4,
}))(View);

const StyledCard = styled(({ theme, cardColor }) => ({
  width: '100%',
  minHeight: 400,
  maxWidth: 420,
  backgroundColor: !cardColor ? theme.lightPrimaryColor : cardColor,
  borderRadius: theme.cardBorderRadius,
  ...Platform.select({
    ios: {
      shadowColor: theme.lightTertiaryColor,
      shadowOffset: {
        width: 0,
        height: 1,
      },
      shadowOpacity: 1,
      shadowRadius: 3,
    },
    android: {
      elevation: 3,
    },
  }),
}))(View);

/*
 * Overflow on iOS, when declared on the same element as a shadow, clips the shadow so it must live
 * on a child wrapper. https://github.com/facebook/react-native/issues/449
 */
const OverflowFix = styled(({ theme }) => ({
  flex: 1,
  borderRadius: theme.cardBorderRadius,
  overflow: 'hidden',
}))(View);

const StyledImage = styled(({ theme }) => ({
  width: undefined,
  height: undefined,
  flex: 1,
  resizeMode: 'cover',
  borderTopRightRadius: theme.cardBorderRadius,
  borderTopLeftRadius: theme.cardBorderRadius,
}))(Image);

const CardTitle = styled(({ theme }) => ({
  flex: 0,
  paddingTop: theme.baseUnit,
  paddingHorizontal: theme.baseUnit,
}))(H4);

const Footer = styled({
  flexDirection: 'row',
  alignItems: 'center',
})(View);

const LikeButton = styled(({ theme }) => ({
  paddingTop: theme.baseUnit / 2,
  paddingHorizontal: theme.baseUnit,
  paddingBottom: theme.baseUnit,
}))(TouchableOpacity);

const MediaCard = enhance(({
  image: imagePath,
  title,
  category,
  style: styleProp = {},
  theme,
  ...otherProps
}) => (
  <CardWrapper>
    <StyledCard
      style={styleProp}
      {...otherProps}
    >
      <OverflowFix>
        <StyledImage source={{ uri: imagePath }} />

        <CardTitle>{startCase(toLower(title))}</CardTitle>

        <Footer>
          <Category type={category} />
          <LikeButton>
            <Icon
              name={'like'}
              size={rem(1.2, theme)}
              fill={theme.baseFontColor}
            />
          </LikeButton>
        </Footer>
      </OverflowFix>
    </StyledCard>
  </CardWrapper>
));

export default MediaCard;
