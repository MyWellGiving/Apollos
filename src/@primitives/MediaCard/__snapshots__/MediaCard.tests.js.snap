// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component should accept a different card color 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
      "marginVertical": 5,
    }
  }
>
  <View
    cardColor="salmon"
    style={
      Object {
        "backgroundColor": "salmon",
        "borderRadius": 6,
        "maxWidth": 420,
        "minHeight": 400,
        "shadowColor": "#dddddd",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 6,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/600/400/?random",
          }
        }
        style={
          Object {
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "flex": 1,
            "height": undefined,
            "resizeMode": "cover",
            "width": undefined,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#505050",
            "flex": 0,
            "fontFamily": null,
            "fontSize": 25.2,
            "fontWeight": "700",
            "lineHeight": 31.28,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        Boom
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M3.8182 5.7778v12.4444H14.2V5.7778H3.8182zM16 6v12c0 1.1046-.8954 2-2 2H4c-1.1046 0-2-.8954-2-2V6c0-1.1046.8954-2 2-2h10c1.1046 0 2 .8954 2 2zm4.1818 2.6952L16 11.8084v.3832l4.1818 3.251V8.695zM16 14.4957V9.4253l6-4.2178v13.585l-6-4.2968z"
              fill="#505050"
            />
          </Svg>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#505050",
                "fontFamily": null,
                "fontSize": 14,
                "lineHeight": 15.87,
                "paddingHorizontal": 10,
              }
            }
          >
            What
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
              fill="#505050"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the Card component should accept and render passed in styles 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
      "marginVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 6,
        "height": 400,
        "maxWidth": 420,
        "minHeight": 400,
        "shadowColor": "#dddddd",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
        "width": "92%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 6,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/600/400/?random",
          }
        }
        style={
          Object {
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "flex": 1,
            "height": undefined,
            "resizeMode": "cover",
            "width": undefined,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#505050",
            "flex": 0,
            "fontFamily": null,
            "fontSize": 25.2,
            "fontWeight": "700",
            "lineHeight": 31.28,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        Boom
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M3.8182 5.7778v12.4444H14.2V5.7778H3.8182zM16 6v12c0 1.1046-.8954 2-2 2H4c-1.1046 0-2-.8954-2-2V6c0-1.1046.8954-2 2-2h10c1.1046 0 2 .8954 2 2zm4.1818 2.6952L16 11.8084v.3832l4.1818 3.251V8.695zM16 14.4957V9.4253l6-4.2178v13.585l-6-4.2968z"
              fill="#505050"
            />
          </Svg>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#505050",
                "fontFamily": null,
                "fontSize": 14,
                "lineHeight": 15.87,
                "paddingHorizontal": 10,
              }
            }
          >
            What
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
              fill="#505050"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`the Card component should render 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
      "marginVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 6,
        "maxWidth": 420,
        "minHeight": 400,
        "shadowColor": "#dddddd",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 6,
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://picsum.photos/600/400/?random",
          }
        }
        style={
          Object {
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "flex": 1,
            "height": undefined,
            "resizeMode": "cover",
            "width": undefined,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#505050",
            "flex": 0,
            "fontFamily": null,
            "fontSize": 25.2,
            "fontWeight": "700",
            "lineHeight": 31.28,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        Boom
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M3.8182 5.7778v12.4444H14.2V5.7778H3.8182zM16 6v12c0 1.1046-.8954 2-2 2H4c-1.1046 0-2-.8954-2-2V6c0-1.1046.8954-2 2-2h10c1.1046 0 2 .8954 2 2zm4.1818 2.6952L16 11.8084v.3832l4.1818 3.251V8.695zM16 14.4957V9.4253l6-4.2178v13.585l-6-4.2968z"
              fill="#505050"
            />
          </Svg>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#505050",
                "fontFamily": null,
                "fontSize": 14,
                "lineHeight": 15.87,
                "paddingHorizontal": 10,
              }
            }
          >
            What
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingBottom": 20,
              "paddingHorizontal": 20,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Svg
            height={21.6}
            viewBox="0 0 24 24"
            width={21.6}
          >
            <Path
              d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
              fill="#505050"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;
